sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -36,4 +56,145 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
       })
       .getMany();
   }
+
+  /**
+   * Save scholastic standing and create new assessment
+   * for an application.
+   * @param applicationId application id.
+   * @param scholasticStandingData scholastic standing data to be saved
+   */
+  async saveScholasticStandingCreateReassessment(
+    applicationId: number,
+    userId: number,
+    scholasticStandingData: ScholasticStandingAPIInDTO,
+  ): Promise<StudentScholasticStanding> {
+    const application = await this.applicationRepo
+      .createQueryBuilder("application")
+      .select(["application", "currentAssessment.id", "offering.id"])
+      .where("application.id = :applicationId", { applicationId })
+      .leftJoin("application.currentAssessment", "currentAssessment")
+      .leftJoin("currentAssessment.offering", "offering")
+      .getOne();
+
+    if (!application) {
+      throw new CustomNamedError(
+        "Application Not found.",
+        APPLICATION_NOT_FOUND,
+      );
+    }
+
+    if (application.applicationStatus !== ApplicationStatus.completed) {
+      throw new CustomNamedError(
+        "Not a completed application.",
+        NOT_A_COMPLETED_APPLICATION,
+      );
+    }
+
+    return this.connection.transaction(async (transactionalEntityManager) => {
+      const now = new Date();
+      const auditUser = {
+        id: userId,
+      } as User;
+
+      // Get existing offering.
+      const existingOffering = await this.offeringRepo
+        .createQueryBuilder("offering")
+        .select(["offering", "educationProgram.id", "institutionLocation.id"])
+        .where("offering.id = :offeringId", {
+          offeringId: application.currentAssessment.offering.id,
+        })
+        .innerJoin("offering.educationProgram", "educationProgram")
+        .innerJoin("offering.institutionLocation", "institutionLocation")
+        .getOne();
+
+      // Create new offering for student scholastic standing.
+      const offering = new EducationProgramOffering();
```

Comment:
For this scenario, can we consider using the spread operator to do the shallow copy of almost all properties and then override what is needed?

Reply:
yup same sense. trying